src/Maestro/Maestro.Data/Helpers/NormalizationOperation.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Maestro.Data.Helpers
+{
+    public class NormalizationOperation
+    {
+        private const string VisualStudioDncEngHost = "dnceng.visualstudio.com";
```

Comment:
We should use DarcLib's URI parsing here and normalize using the returned info..  We should not hardcode hosts.

Reply:
I tried finding an existing parser but could not find one, hence the class I created. Could you point me to it please?

Reply:
https://github.com/dotnet/arcade-services/blob/master/src/Microsoft.DotNet.Darc/src/DarcLib.AzDev/AzureDevOpsClient.cs#L769

Reply:
Ok, although if we are using that I'd need to remove the TODO which is currently there since we'll rely on that function to remove the user from the URL.

Reply:
then maybe just use that code over here? or factor it out to have a version that allows for alternate regex and a version that allows for only a strict set?